File: app/src/main/java/com/example/water/cproject/ListCodeAdapter.java
package com.example.water.cproject;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by watering on 18. 3. 17.
 */

public class ListCodeAdapter extends BaseAdapter {
    private final List<String> list;
    private final LayoutInflater inflater;

    public ListCodeAdapter(Context context, ArrayList<String> data) {
        this.list = data;
        inflater = LayoutInflater.from(context);
    }

    @Override
    public int getCount() {
        return this.list.size();
    }

    @Override
    public Object getItem(int position) {
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if(convertView == null) {
            convertView = inflater.inflate(R.layout.layout_list, parent, false);
        }

        TextView code = convertView.findViewById(R.id.textView_layout_list);
        code.setText(list.get(position).toString());
        return convertView;
    }
}

File: app/src/main/java/com/example/water/cproject/ListCodeAdapter.kt
package com.example.water.cproject

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import java.util.*

/**
 * Created by watering on 18. 3. 17.
 */

class ListCodeAdapter(context: Context, data: ArrayList<String>?) : BaseAdapter() {
    private val list: List<String>?
    private val inflater: LayoutInflater

    init {
        this.list = data
        this.inflater = LayoutInflater.from(context)
    }

    override fun getCount(): Int {
        return this.list!!.size
    }
    override fun getItem(position: Int): Any {
        return list!![position]
    }
    override fun getItemId(position: Int): Long {
        return 0
    }
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view: View?

        if (convertView == null) {
            view = this.inflater.inflate(R.layout.layout_list, parent, false)
        }
        else {
            view = convertView
        }

        val code = view!!.findViewById(R.id.textView_layout_list) as TextView
        if (!list!!.isEmpty()) code.text = list[position]
        return view
    }
}
